src/app/ReadClient.cpp

```diff
@@ -485,6 +485,15 @@ CHIP_ERROR ReadClient::GenerateDataVersionFilterList(DataVersionFilterIBs::Build
 void ReadClient::OnActiveModeNotification()
 {
     VerifyOrDie(mpImEngine->InActiveReadClientList(this));
+
+    // Note: this API only works when issuing subscription via SendAutoResubscribeRequest. When SendAutoResubscribeRequest is
+    // called, either mEventPathParamsListSize or mAttributePathParamsListSize is not 0.
+    VerifyOrReturn(mReadPrepareParams.mEventPathParamsListSize != 0 || mReadPrepareParams.mAttributePathParamsListSize != 0);
+
+    // If mCatsMatchCheckIn is true, it means cats used in icd registration matches with the one in current subscription,
```

Comment:
Matches _which side_?  Should be "our side of the current subscription" or something, right?

But do we really want to push this off onto the API client?  Or just examine the fabric table?

And if we do push this off onto the API client, as this PR does, then should this boolean just be `mResetSubscriptionOnCheckin`?

Reply:
If we are going with fabrictable path, then we can get fabricTable from interaction model engine, then use get fabricId via FabricIndex readClient::GetFabricIndex() const { return mPeer.GetFabricIndex(); },  further we can get cats via FetchCATs(mFabricIndex, cats).

Reply:
this logic has been removed, we are leaning toward examining the fabric table, thanks